spring-kafka/src/main/java/org/springframework/kafka/listener/ConcurrentMessageListenerContainer.java

```diff
@@ -204,14 +206,20 @@ public boolean isContainerPaused() {
 
 	@Override
 	public boolean isChildRunning() {
-		if (!isRunning()) {
-			return false;
-		}
-		for (MessageListenerContainer container : this.containers) {
-			if (container.isRunning()) {
+		this.lifecycleLock.lock();
+		try {
+			for (MessageListenerContainer container : this.containers) {
+				if (container.isRunning()) {
+					return true;
+				}
+			}
+			if ((!isRunning() && this.startedContainers.get() > 0)) {
```

Comment:
I don't understand this extra logic.
The purpose of the `isChildRunning()` is to report if any child container is active at the moment.
Why do we need to check `!this.isRunning()` and then number of started containers?
Didn't we discuss with your in other PR (#3377 ) that it is abnormal to have child container running and stopped parent one?

Reply:
The condition in the Line:216 is the critical logic. `running` status of the ConcurrentContainer will be set to `false` for the following conditions.

1) ConcurrentContainer itself is stopped manually.
2) One of the child container is stopped for any reason.
 
`startedContainers` count will be decremented only when the actual container exits. So, i have verified if the `ConcurrentContainer` is stopped and it is having any child containers processing still messages. In this case, it should return `true`.